WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -0,0 +1,618 @@
+            where searchText?.characters.count == 0 && recentlyTrashedPostObjectIDs.count > 0 {
+            
+            let trashedPredicate = NSPredicate(format: "SELF IN %@", recentlyTrashedPostObjectIDs)
+            
+            if let originalFilterPredicate = filterPredicate {
+                filterPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [originalFilterPredicate, trashedPredicate])
+            } else {
+                filterPredicate = trashedPredicate
+            }
+        }
+        
+        if let filterPredicate = filterPredicate {
+            predicates.append(filterPredicate)
+        }
+        
+        if let searchText = searchText where searchText.characters.count > 0 {
+            let searchPredicate = NSPredicate(format: "postTitle CONTAINS[cd] %@", searchText)
+            predicates.append(searchPredicate)
+        }
+        
+        let predicate = NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        return predicate
+    }
+    
+    // MARK: - Table View Handling
+
+    func sectionNameKeyPath() -> String {
+        return NSStringFromSelector(#selector(Page.sectionIdentifier))
+    }
+    
+    func tableView(tableView: UITableView, estimatedHeightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return CGFloat(self.dynamicType.pageCellEstimatedRowHeight)
+    }
+    
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        
+        if let page = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? Page {
+            if cellIdentifierForPage(page) == self.dynamicType.restorePageCellIdentifier {
+                return CGFloat(self.dynamicType.pageCellEstimatedRowHeight)
+            }
+            
+            let width = CGRectGetWidth(tableView.bounds)
+            return self.tableView(tableView, heightForRowAtIndexPath: indexPath, forWidth: width)
+        } else {
+            return 0
+        }
+    }
+    
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath, forWidth width: CGFloat) -> CGFloat {
+        configureCell(cellForLayout, atIndexPath: indexPath)
+        let size = cellForLayout.sizeThatFits(CGSizeMake(width, CGFloat.max))
+        let height = ceil(size.height)
+        
+        return height
+    }
+    
+    func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return CGFloat(self.dynamicType.pageSectionHeaderHeight)
+    }
+    
+    func tableView(tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return CGFloat.min
+    }
+    
+    func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView! {
+        let sectionInfo = tableViewHandler?.resultsController.sections?[section]
+        let nibName = NSStringFromClass(PageListSectionHeaderView.self)
```

Comment:
Instead of '''NSStringFromClass(PageListSectionHeaderView.self)''' you can write just this: '''String(PageListSectionHeaderView)'''